development/webpack/utils/config.ts

```diff
@@ -0,0 +1,186 @@
+import { join } from 'node:path';
+import { readFileSync } from 'node:fs';
+import { parse as parseYaml } from 'yaml';
+import { parse } from 'dotenv';
+import { setEnvironmentVariables } from '../../build/set-environment-variables';
+import type { Variables } from '../../lib/variables';
+import { type Args } from './cli';
+import { getMetaMaskVersion } from './helpers';
+
+const BUILDS_YML_PATH = join(__dirname, '../../../builds.yml');
+
+/**
+ * Coerce `"true"`, `"false"`, and `"null"` to their respective JavaScript values.
+ *
+ * @param value
+ * @returns
+ */
+function coerce(value: string) {
+  if (value === 'true') return true;
+  if (value === 'false') return false;
+  if (value === 'null') return null;
+  return value;
+}
+
+/**
+ * @returns The definitions loaded from process.env.
+ */
+function loadEnv(): Map<string, unknown> {
+  const definitions = new Map<string, unknown>();
+  Object.entries(process.env).forEach(([key, value]) => {
+    if (typeof value === 'undefined') return;
+    definitions.set(key, coerce(value));
+  });
+  return definitions;
+}
+
+/**
+ * @param definitions
+ * @param rcFilePath - The path to the rc file.
+ */
+function addRc(definitions: Map<string, unknown>, rcFilePath: string): void {
+  try {
+    const rc = parse(readFileSync(rcFilePath, 'utf8'));
+    Object.entries(rc).forEach(([key, value]) => {
+      if (definitions.has(key)) return;
+      definitions.set(key, coerce(value));
+    });
+  } catch {
+    // ignore
```

Comment:
We don't care if the `.metamaskrc` file doesn't exist, we can ignore errors reading it.